src/sections/settings/token-settings/TokenSettings.jsx

```diff
@@ -0,0 +1,215 @@
+  const [isEditing, setEditing] = useState(false);
+  const [localRatio, setLocalRatio] = useState(ratio);
+  const [transactionState, setTransactionState] = useState("");
+  const [isVerifingTransaction, setVerifingTransaction] = useState(null);
+
+  const { setERC1155, isFetching, isLoading } = useSetERC1155({
+    erc1155Address: address,
+    ercId: id,
+    ratio: localRatio,
+  });
+
+  useEffect(() => {
+    setLocalRatio(ratio);
+  }, [ratio]);
+
+  useEffect(() => {
+    fetchAddress();
+    fetchId();
+    fetchRatio();
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  function handleEditClick(shouldEdit) {
+    setEditing(shouldEdit);
+  }
+
+  function handleRatioChange(event) {
+    setLocalRatio(event.target.valueAsNumber);
+  }
+
+  function handleDecline() {
+    setLocalRatio(ratio);
+  }
+
+  async function handleTransactionSuccess(transaction) {
+    setTransactionState("success");
+    setVerifingTransaction(true);
+    await transaction.wait();
+    setVerifingTransaction(false);
+    fetchRatio();
+  }
+
+  async function handleTransactionError(error) {
+    console.error(error);
+    setTransactionState("error");
+  }
+
+  function handleSnackbarClose() {
+    setTransactionState("");
+  }
+
+  async function handleSave() {
+    setERC1155({
+      erc1155Address: address,
+      ercId: id,
+      ratio: localRatio,
+      onSuccess: handleTransactionSuccess,
+      onError: handleTransactionError,
+    });
+  }
+
+  const showSpinner =
+    isFetchingAddress ||
+    isLoadingAddress ||
+    isFetchingId ||
+    isLoadingId ||
+    isFetchingRatio ||
+    isLoadingRatio;
+
+  if (showSpinner) {
+    return (
+      <Box textAlign="center" mt={2}>
+        <CircularProgress />
+      </Box>
+    );
+  }
+
+  return (
+    <Box ml={5}>
+      <Grid
+        container
+        justifyContent="space-between"
+        alignItems="center"
+        wrap="nowrap"
+        color={theme.palette.grey["600"]}
+        sx={{ borderWidth: "1px 0 1px 0", minHeight: 98 }}
+        border={`1px solid ${theme.palette.divider}`}
+        py={3}
+        px={2}
+      >
+        {!!isEditing ? (
+          <>
+            <Grid item sm={12}>
+              <TextField
+                type="number"
+                value={String(localRatio)}
+                onChange={handleRatioChange}
+                variant="standard"
+                label="Amount of tokens to burn"
+                sx={{ width: "90%" }}
+                autoFocus
+              />
+            </Grid>
+            <Grid item>
+              <Tooltip title="Save">
+                <IconButton onClick={() => handleEditClick(false)}>
+                  <Iconify icon="charm:tick" />
+                </IconButton>
+              </Tooltip>
+            </Grid>
+          </>
+        ) : (
+          <>
+            <Grid item sm={10}>
+              <Typography variant="subtitle1">
+                Tokens to burn for receiving an NFT
```

Comment:
This should be named Burn Ratio, will be know to admin what it means

Reply:
Added a fix for that :)